databricks/koalas/indexes.py

```diff
@@ -174,7 +174,8 @@ def names(self, names: List[Union[str, Tuple[str, ...]]]) -> None:
         if len(internal.index_map) != len(names):
             raise ValueError('Length of new names must be {}, got {}'
                              .format(len(internal.index_map), len(names)))
-        names = [name if isinstance(name, tuple) else (name,) for name in names]
+        if None not in names:
+            names = [name if isinstance(name, tuple) else (name,) for name in names]
```

Comment:
@itholic, does it work if some of names are only `None`? E.g.)

```python
pd.MultiIndex.from_tuples([('a', 'b'), ('a', 'c')], names=['first', None]).names
```

Reply:
@HyukjinKwon , thanks for the comment!

i missed that case 😢 , and just added it!